nixos/modules/services/web-servers/meguca.nix

```diff
@@ -0,0 +1,123 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.meguca;
+  postgres = config.services.postgresql;
+in
+{
+  options.services.meguca = {
+    enable = mkEnableOption "meguca";
+
+    baseDir = mkOption {
+      type = types.path;
+      default = "/var/lib/meguca";
+      description = "Location where meguca stores it's database and links.";
+    };
+
+    password = mkOption {
+      type = types.str;
+      default = "meguca";
+      description = "Password for the meguca database.";
```

Comment:
The password is stored in clear text in nix readable for all users. We usually have at least an alternative option that stored the password in a file outside the nix store: https://github.com/NixOS/nixpkgs/issues/24288
Btw where is this password used in the service?

Reply:
If meguca also accepts unix sockets for database connections, authentication could be also done via user without any password.
I would not hardcode `meguca` as default password and rather let the user choose.

Reply:
`meguca` is hard coded into upstream itself as the default, however with `postgresArgs` you can define the password to access the database with, in which is [set each service start](https://github.com/Chiiruno/nixpkgs/blob/614b7d5a13b4414caeb9628c2be23f776b6463cc/nixos/modules/services/web-servers/meguca.nix#L88-L91).

Reply:
Ah, so it's world-readable.
A password file in /var/meguca (or user-definable) would be ideal.